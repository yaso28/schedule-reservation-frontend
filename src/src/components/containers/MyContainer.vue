<template>
  <!-- post processing cover -->
  <MyProcessing
    id="processing-post"
    :processing="processingPost"
    :z-index="9999"
    class="c-app"
  >
    <!-- sidebar -->
    <MySidebar />
    <CWrapper>
      <!-- header -->
      <MyHeader />
      <!-- body -->
      <!-- get processing cover -->
      <MyProcessing
        id="processing-get"
        :processing="processingGet"
        class="c-body"
      >
        <main class="c-main">
          <CContainer fluid>
            <!-- alert -->
            <div id="alert-area">
              <CAlert
                v-for="alertValue in alertValues"
                :key="alertValue.key"
                :color="alertValue.type"
                close-button
              >
                {{ $t(alertValue.message) }}
              </CAlert>
            </div>

            <!-- main -->
            <MyPage />
          </CContainer>
        </main>
      </MyProcessing>
    </CWrapper>
  </MyProcessing>
</template>

<script>
import { mapState } from 'vuex'
export default {
  name: 'MyContainer',
  components: {
    MyProcessing: require('@/components/wrappers/MyProcessing').default,
    MySidebar: require('./MySidebar').default,
    MyHeader: require('./MyHeader').default,
    MyPage: require('./MyPage').default
  },
  inheritAttrs: false,
  computed: {
    ...mapState({
      processingGet: state => state.processing.get,
      processingPost: state => state.processing.post,
      alertValues: state => state.alert.values
    })
  }
}
</script>
