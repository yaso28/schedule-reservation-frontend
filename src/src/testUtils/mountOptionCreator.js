import { createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import routes from '@/router/routes'
import Vuex from 'vuex'
import VueI18n from 'vue-i18n'
import i18nOptions from '@/i18n/options'
import VueMeta from 'vue-meta'
import CoreuiVue from '@coreui/vue'
import icons from '@/icons'
import services from '@/services'

class MountOptionCreator {
  create (storeModules = null) {
    const localVue = createLocalVue()

    // router
    localVue.use(VueRouter)
    const router = new VueRouter({
      mode: 'history',
      base: process.env.BASE_URL,
      routes
    })

    // i18n
    localVue.use(VueI18n)
    const i18n = new VueI18n(i18nOptions)

    // vue-meta
    localVue.use(VueMeta)

    // Core UI
    localVue.use(CoreuiVue)

    const option = {
      localVue,
      router,
      i18n,
      icons,
      provide: services
    }

    // store
    if (storeModules) {
      localVue.use(Vuex)
      option.store = new Vuex.Store({
        modules: storeModules
      })
    }

    return option
  }
}

export default new MountOptionCreator()
