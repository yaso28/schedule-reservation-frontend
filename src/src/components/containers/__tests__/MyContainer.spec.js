import { mount } from '@vue/test-utils'
import mountOptionCreator from '@/testUtils/mountOptionCreator'
import MyContainer from '../MyContainer'
import storeModules from '@/store/modules'

jest.mock('@/store/modules')

describe('MyContainer', () => {
  let store
  let mountOption
  let wrapper
  beforeEach(() => {
    mountOption = mountOptionCreator.create(storeModules)
    mountOption.stubs = ['MySidebar', 'MyHeader', 'MyPage']
    store = mountOption.store
    store.state.processing.get = false
    store.state.processing.post = false
    store.state.alert.values = []
  })

  afterEach(() => {
    jest.resetAllMocks()
  })

  describe('通常', () => {
    beforeEach(() => {
      wrapper = mount(MyContainer, mountOption)
    })

    test('表示', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('processing-cover', () => {
    describe('Get処理中', () => {
      beforeEach(async () => {
        store.state.processing.get = true
        wrapper = mount(MyContainer, mountOption)
      })

      test('Getカバー表示', () => {
        expect(wrapper.find('#processing-get').attributes()).toMatchSnapshot()
      })
    })

    describe('Post処理中', () => {
      beforeEach(async () => {
        store.state.processing.post = true
        wrapper = mount(MyContainer, mountOption)
      })

      test('Postカバー表示', () => {
        expect(wrapper.find('#processing-post').attributes()).toMatchSnapshot()
      })
    })
  })

  describe('アラート', () => {
    beforeEach(async () => {
      store.state.alert.values = [
        { key: 1, type: 'success', message: 'message.save.success' },
        { key: 2, type: 'danger', message: 'message.error.api' },
        { key: 3, type: 'danger', message: 'message.error.validation' }
      ]
      wrapper = mount(MyContainer, mountOption)
    })

    test('アラート表示', () => {
      expect(wrapper.find('#alert-area').element).toMatchSnapshot()
    })
  })
})
