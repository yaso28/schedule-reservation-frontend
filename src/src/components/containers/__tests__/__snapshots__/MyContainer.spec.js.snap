// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyContainer processing-cover Get処理中 Getカバー表示 1`] = `
Object {
  "id": "processing-get",
  "style": "top: 0px; bottom: 0px; left: 0px; right: 0px; position: absolute; z-index: 900;",
}
`;

exports[`MyContainer processing-cover Post処理中 Postカバー表示 1`] = `
Object {
  "id": "processing-post",
  "style": "top: 0px; bottom: 0px; left: 0px; right: 0px; position: absolute; z-index: 9999;",
}
`;

exports[`MyContainer アラート アラート表示 1`] = `
<div
  id="alert-area"
>
  <transition-stub
    appear="true"
    data-v-c95d1f08=""
    name="fade"
  >
    <div
      aria-atomic="true"
      aria-live="polite"
      class="alert alert-dismissible alert-success"
      data-v-c95d1f08=""
      role="alert"
    >
      <button
        aria-label="Close"
        class="close"
        data-v-c95d1f08=""
        type="button"
      >
        ×
      </button>
      
              保存しました。
            
    </div>
  </transition-stub>
  <transition-stub
    appear="true"
    data-v-c95d1f08=""
    name="fade"
  >
    <div
      aria-atomic="true"
      aria-live="polite"
      class="alert alert-dismissible alert-danger"
      data-v-c95d1f08=""
      role="alert"
    >
      <button
        aria-label="Close"
        class="close"
        data-v-c95d1f08=""
        type="button"
      >
        ×
      </button>
      
              予期せぬエラーが発生しました。
            
    </div>
  </transition-stub>
  <transition-stub
    appear="true"
    data-v-c95d1f08=""
    name="fade"
  >
    <div
      aria-atomic="true"
      aria-live="polite"
      class="alert alert-dismissible alert-danger"
      data-v-c95d1f08=""
      role="alert"
    >
      <button
        aria-label="Close"
        class="close"
        data-v-c95d1f08=""
        type="button"
      >
        ×
      </button>
      
              入力項目に誤りがあります。
            
    </div>
  </transition-stub>
</div>
`;

exports[`MyContainer 通常 表示 1`] = `
<div
  class="position-relative c-app"
>
  <mysidebar-stub />
   
  <div
    class="c-wrapper"
  >
    <myheader-stub />
     
    <div
      class="position-relative c-body"
    >
      <main
        class="c-main"
      >
        <div
          class="container-fluid"
        >
          <div
            id="alert-area"
          />
           
          <mypage-stub />
        </div>
      </main>
       
      <div
        id="processing-get"
        style="top: 0px; bottom: 0px; left: 0px; right: 0px; position: absolute; z-index: 900; display: none;"
      >
        <div
          style="position: absolute; top: 50%; left: 50%; transform: translateX(-50%) translateY(-50%);"
        >
          <div
            aria-hidden="false"
            aria-label="Loading"
            class="spinner-border text-success"
            role="status"
          />
        </div>
      </div>
    </div>
  </div>
   
  <div
    id="processing-post"
    style="top: 0px; bottom: 0px; left: 0px; right: 0px; position: absolute; z-index: 9999; display: none;"
  >
    <div
      style="position: absolute; top: 50%; left: 50%; transform: translateX(-50%) translateY(-50%);"
    >
      <div
        aria-hidden="false"
        aria-label="Loading"
        class="spinner-border text-success"
        role="status"
      />
    </div>
  </div>
</div>
`;
